dashboard/lib/level_loader.rb

```diff
@@ -48,7 +48,11 @@ def self.import_levels(level_file_glob)
       changed_lcds = changed_levels.map(&:level_concept_difficulty).compact
       lcd_update_columns = LevelConceptDifficulty.columns.map(&:name).map(&:to_sym).
           reject {|column| %i{id level_id created_at}.include? column}
-      LevelConceptDifficulty.import! changed_lcds, on_duplicate_key_update: lcd_update_columns
+      if rack_env?(:development)
+        LevelConceptDifficulty.import changed_lcds, on_duplicate_key_update: lcd_update_columns
+      else
+        LevelConceptDifficulty.import! changed_lcds, on_duplicate_key_update: lcd_update_columns
```

Comment:
I don't fully understand the issue yet. The contributor was missing encrypted levels because they didn't have the `properties_encryption_key` for them, which should be fine - we want it to be possible to build and seed a local instance without that key.  However, LevelConceptDifficulty here is seeded from the set of changed levels imported and it seems like a different problem if unimported levels are included in that set.

Do you have repro steps, or more details on the exception raised when the contributor tried to seed levels?

Reply:
Here is the error she got:
```
indhu@indhu-VirtualBox:~/code-dot-org$ bundle exec rake install --trace
** Invoke install (first_time)
** Invoke install:all (first_time)
** Invoke install:hooks (first_time)
** Execute install:hooks
** Invoke install:locals_yml (first_time)
** Execute install:locals_yml
** Invoke install:apps (first_time)
** Execute install:apps
which npm
** Invoke install:dashboard (first_time)
** Execute install:dashboard
bundle --without production adhoc staging test levelbuilder integration --quiet --jobs 1
mysql://root@localhost/dashboard_development
RAILS_ENV=development RACK_ENV=development bundle exec rake dashboard:setup_db
rake aborted!
'RAILS_ENV=development RACK_ENV=development bundle exec rake dashboard:setup_db' returned 1
rake aborted!
ActiveRecord::RecordInvalid: Validation failed: Level is required
[CDO]/dashboard/lib/level_loader.rb:51:in `block in import_levels'
[CDO]/dashboard/lib/level_loader.rb:27:in `import_levels'
[CDO]/dashboard/lib/level_loader.rb:7:in `load_custom_levels'
[CDO]/dashboard/lib/tasks/seed.rake:214:in `block (2 levels) in <top (required)>'
[BIN]/bundle:23:in `load'
[BIN]/bundle:23:in `<main>'
Tasks: TOP => dashboard:setup_db => seed:all => seed:scripts => seed:custom_levels
(See full trace by running task with --trace)
Model files unchanged.
WARNING: level 'Allthethings Encrypted Play Lab Level' not seeded properly due to missing CDO.properties_encryption_key
WARNING: level 'U3L6 CSP CFU Sequential vs Event Driven' not seeded properly due to missing CDO.properties_encryption_key
WARNING: level 'U3L7 CSP CFU What was confusing' not seeded properly due to missing CDO.properties_encryption_key
Tasks: TOP => install => install:all => install:dashboard
```

I can also wait for Dave to get back before shipping this. He was the one that suggested the change on this line

Reply:
Interesting, so it is the validation on LevelConceptDifficulty that the associated level exists that is failing.

My instinct is that instead of a special case for development here, we want to change the `changed_levels` search on line 41 to actually omit levels it can't load from the database, since prior to that we've already got a stub import that should have created levels before we reached this step.  Can we take a look together sometime this week?

Reply:
Yes that sounds great. I put time on your calendar for tomorrow morning but if that doesn't work lets find another time.

Reply:
I like @islemaster 's suggestion -- after line 41, you can do something like this to filter out encrypted levels:
```ruby
changed_levels.reject!(&:encrypted) unless CDO.properties_encryption_key
```